packages/tasks/src/hardware.ts

```diff
@@ -0,0 +1,305 @@
+/**
+ * Biden AI Executive Order
+ * https://www.whitehouse.gov/briefing-room/presidential-actions/2023/10/30/executive-order-on-the-safe-secure-and-trustworthy-development-and-use-of-artificial-intelligence/
+ */
+export const TFLOPS_THRESHOLD_WHITE_HOUSE_MODEL_TRAINING_TOTAL = 10 ** 14;
+export const TFLOPS_THRESHOLD_WHITE_HOUSE_MODEL_TRAINING_TOTAL_BIOLOGY = 10 ** 11;
+export const TFLOPS_THRESHOLD_WHITE_HOUSE_CLUSTER = 10 ** 8;
+
+/**
+ * EU AI Act
+ * https://ec.europa.eu/commission/presscorner/detail/en/qanda_21_1683
+ */
+export const TFLOPS_THRESHOLD_EU_AI_ACT_MODEL_TRAINING_TOTAL = 10 ** 13;
+
+export interface HardwareSpec {
+	/**
+	 * Approximate value, in FP32.
+	 * This is only approximate/theoretical and shouldn't be taken too seriously.
+	 * Currently the CPU values are from cpu-monkey.com
+	 * while the GPU values are from techpowerup.com
+	 *
+	 * Note to reviewers: I got fed up with data entry,
+	 * and HuggingChat running Llama3 with Web search was failing a bit,
+	 * so some of those values are kinda random. Forgive me and please feel free to improve.
+	 */
+	tflops: number;
+	/**
+	 * If an array is specified, options of memory size (can be VRAM, unified RAM)
+	 * e.g. an A100 exists in 40 or 80 GB.
+	 */
+	memory?: number[];
+}
+
+export const DEFAULT_MEMORY_OPTIONS = [8, 16, 24, 32, 40, 48, 64, 80];
+
+export const SKUS = {
+	GPU: {
+		NVIDIA: {
+			H100: {
+				tflops: 51.22,
+				memory: [80],
+			},
+			L40: {
+				tflops: 90.52,
+				memory: [24],
+			},
+			A100: {
+				tflops: 19.49,
+				memory: [80, 40],
+			},
+			A40: {
+				tflops: 37.42,
+				memory: [48],
+			},
+			A10: {
+				tflops: 31.24,
+				memory: [24],
+			},
+			T4: {
+				tflops: 8.141,
+				memory: [16],
+			},
+			"RTX 4090": {
+				tflops: 8.58,
+				memory: [24, 16],
+			},
+			"RTX 4080 SUPER": {
+				tflops: 8.58,
+				memory: [16],
+			},
+			"RTX 4080": {
+				tflops: 8.58,
+				memory: [12],
+			},
+			"RTX 4070": {
+				tflops: 8.58,
+				memory: [12, 8],
+			},
+			"RTX 4070 Ti": {
+				tflops: 8.58,
+				memory: [12],
+			},
+			"RTX 4070 Super": {
+				tflops: 8.58,
+				memory: [12],
+			},
+			"RTX 4070 Ti Super": {
+				tflops: 7.57,
+				memory: [16],
+			},
+			"RTX 3090": {
+				tflops: 7.57,
```

Comment:
```suggestion
				tflops: 35.58,
```

Source: https://news.ycombinator.com/item?id=34261797#:~:text=The%20RTX%203090%20has%2035.58,http...%20%7C%20Hacker%20News&text=johndough%20on%20Jan%205%2C%202023,performance%20on%20a%20sin...